packages/elements/image-ui/src/ImageElement/ImageElement.tsx

```diff
@@ -1,31 +1,70 @@
 import * as React from 'react';
-import { useFocused, useSelected } from 'slate-react';
+import { setNodes } from '@udecode/plate-common';
+import { useEditorRef } from '@udecode/plate-core';
+import { Resizable } from 're-resizable';
+import { ReactEditor, useFocused, useSelected } from 'slate-react';
 import { getImageElementStyles } from './ImageElement.styles';
 import { ImageElementProps } from './ImageElement.types';
 
+const { useCallback } = React;
+
 export const ImageElement = (props: ImageElementProps) => {
   const { attributes, children, element, nodeProps } = props;
 
-  const { url } = element;
+  const { url, width, caption = { text: '' } } = element;
   const focused = useFocused();
   const selected = useSelected();
+  const editor = useEditorRef();
 
   const styles = getImageElementStyles({ ...props, focused, selected });
 
+  const onChangeImgWidth = useCallback(
+    (_width: number) => {
+      const path = ReactEditor.findPath(editor, element);
+      setNodes(editor, { width: _width }, { at: path });
+    },
+    [editor, element]
+  );
+
+  const handleChangeCaption: React.ChangeEventHandler<HTMLInputElement> = (
+    e
+  ) => {
+    const path = ReactEditor.findPath(editor, element);
+    setNodes(
+      editor,
+      { caption: { ...caption, text: e.target.value } },
+      { at: path }
+    );
+  };
+
   return (
     <div
       {...attributes}
       css={styles.root.css}
       className={styles.root.className}
     >
       <div contentEditable={false}>
-        <img
-          data-testid="ImageElementImage"
-          css={styles.img?.css}
-          className={styles.img?.className}
-          src={url}
-          alt=""
-          {...nodeProps}
+        <Resizable
+          maxWidth="100%"
+          defaultSize={{ width: width ?? '100%', height: '100%' }}
+          resizeRatio={1.5}
+          lockAspectRatio
+          onResizeStop={(e, e1, e2) => onChangeImgWidth(e2.offsetWidth)}
+        >
+          <img
+            data-testid="ImageElementImage"
+            css={styles.img?.css}
+            className={styles.img?.className}
+            src={url}
+            alt={caption.text}
+            {...nodeProps}
+          />
+        </Resizable>
+        <input
+          value={caption.text}
+          onChange={handleChangeCaption}
+          css={styles.captionInput?.css}
+          placeholder="Write a caption"
```

Comment:
In the future we can replace this by a slate editor to support multi-lines captions

Reply:
We can do it now. If we add data in production then we need to support for backward compatibility as well.